### PR TITLE
Boot: write Apploader version into memory

### PR DESCRIPTION
**Pull request by Tilka**
When HLE-ing the apploader, this is necessary to avoid `Error #002` errors. Homebrew doesn't care.
xfb1_n000000_640x480

### CODE REVIEW

**Comment by Pokechu22** [comment]
What games does this affect? It feels like I'm missing context as to why everything wasn't broken beforehand.

I'm also confused as to why this is on its own instead of in `SetupWiiMemory`, which seems like a more sensible place. I'm probably missing something.

**Comment by BhaaLseN** [comment]
> I'm probably missing something.

This writes the IOS version, which is booted _after_ `SetupWiiMemory` (so it can't be inside unless we also move the call to `BootIOS`.)

**Comment by Tilka** [comment]
This is not the boot path for discs. This is booting individual executables (dol/elf) which you usually only do for homebrew, but you can find various leftovers on game discs. In this case I was interested in ncdconfigtool.elf from Dood's Big Adventure.

I didn't put it into SetupWiiMemory / BootIOS because those are supposed to emulate boot2 / IOS, but here we're emulating an effect of the apploader. In practice it probably doesn't matter, real apploaders would just overwrite the value.

**Review by JosJuice — APPROVED** [review]

**Comment by Pokechu22** [comment]
OK, and I assume this is *also* not used when booting an extracted disc image (since that gets treated as if it were a full disc)? I remember trying to boot separate elf files and it didn't work even when I had the disc set as inserted by default, and I guess this is why.

**Comment by Tilka** [comment]
Yes, exactly.

**Comment by JosJuice** [comment]
@Tilka When there is an active discussion, I would suggest not merging your own PR as soon as you have one approval. Better to give everyone a chance to weigh in first.

**Pull request merged**
